git_machete/github.py

```diff
@@ -21,12 +23,63 @@ def __repr__(self) -> str:
         return f"PR #{self.number} by {self.user}: {self.head} -> {self.base}"
 
 
-GITHUB_TOKEN_ENV_VAR = 'GITHUB_TOKEN'
+GITHUB_TOKEN_ENV_VAR = "GITHUB_TOKEN"
 
+# GitHub Enterprise deployments use alternate domains.
+# The logic in this module will need to be expanded to detect
+# and use alternate remote domains to provide enterprise support.
+GITHUB_DOMAIN = "github.com"
+GITHUB_REMOTE_PATTERNS = [
+    "^https://github\\.com/(.*)/(.*)\\.git$",
+    "^git@github\\.com:(.*)/(.*)\\.git$",
+]
 
-def github_token() -> Optional[str]:
+
+def _token_from_gh() -> Optional[str]:
+
+    # Abort without error if `gh` isn't available
+    gh = shutil.which('gh')
+    if not gh:
+        return None
+
+    # Run via subprocess.run as we're insensitive to return code.
+    #
+    # `gh` can store auth token for public and enterprise domains, specify
+    # single domain for lookup.
+    # This is *only* github.com until enterprise support is added.
+    proc = subprocess.run(
+        [gh, "auth", "status", "--hostname", GITHUB_DOMAIN, "--show-token"],
```

Comment:
Okay... for clarity, if `gh` fails for a reason different than `gh` not being available in the path (e.g. old version which doesn't have `auth` subcommand; or user not logged yet)... then no exception is thrown here right?

Reply:
I guess in such case the regex below won't match anything of course anyway...

Reply:
Oh ok, `insensitive to return code` sorry didn't notice

Reply:
Yes, subprocess.run just returns the CompletedProcess.
You can raise via https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.check_returncode